src/index.ts

```diff
@@ -0,0 +1,89 @@
+import { RefObject, useEffect, useRef, useState } from 'react';
+
+export interface Options {
+  accept?: string;
+  multiple?: boolean;
+}
+
+export interface OnChange<T extends Event = Event> {
+  (event: T): void;
+}
+
+export type Return = {
+  ref: RefObject<HTMLInputElement>;
+  files: File[] | null;
+};
+
+function useInputFile(
+  args: {
+    ref?: RefObject<HTMLInputElement>;
+    options?: Options | undefined;
+    onChange?: OnChange;
+  } = {}
+): Return {
+  const [files, setFiles] = useState<File[] | null>(null);
+  const defaultRef = useRef<HTMLInputElement>(null);
+  const customHandler = useRef<OnChange | undefined>(undefined);
+
+  const ref = args.ref || defaultRef;
+  customHandler.current = args.onChange;
+
+  useEffect(() => {
+    if (ref.current) {
+      ref.current.setAttribute('type', 'file');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
maybe leave `ref` in there? it's not going to be picked up by the diff at least without the eslint disable.

Reply:
OK 👍